### ISSUE TITLE
Error: reading body number of exported fields (3) on response type doesnt match expectation (1)

### ISSUE THREAD

**Issue by graugans** [alexejk/go-xmlrpc#75]
First of all tahnk you very much for your great work. I have an weird XML-RPC server implementation which does response with a different amount of values in case of an internal error occurred, hence the message 

```
Error: reading body number of exported fields (3) on response type doesnt match expectation (1)
```

Is there an option to omit the struct value in case it is empty? Maybe something like this:

```go
result := &struct {
     Error   int                                                                                                                                                            
     Version string `xmlrpc:"Version,omitempty"`                                                                                                                              
     Type    string `xmlrpc:"Type,omitempty"`
}{}
```
In case of a successful operation the returned data from the device is `{0,"4711","AA"}` and in case of an error only `{-100}` is returned.

**Comment by alexejk** [comment]
Hi @graugans, I haven't seen a case like this but i think it should be possible to implement. I'll take a look at this

**Comment by alexejk** [comment]
I've given it a quick stab based on the scarce information i have right now. You can give 0.5.0-RC1 a try when you have a moment, i will make a proper release if this works or after some fixes.

**Comment by alexejk** [comment]
Okay so your  response contains a single array value, so that should just work by specifying that your response is expected to be an array. This would not require any modifications.

```
type MyResponse struct {
	Response []any
}
```

**However** I've noticed that your system does NOT wrap all values in data types. First parameter is correctly wrapped into <i4> which identifies the integer. Problem is that second and third parameter in the array should be wrapped into <i4> or <string> but they aren't. 

Today the library will return nil on them...
I will look into changing implementation to have the flag that would default unknown types to string (or attempt at least) but that will likely come at certain tradeoffs.. Will get back to you on this.

**Comment by graugans** [comment]
Yeah, the device is not behaving in an optimal way. Anyway according to the specification here: http://xmlrpc.com/spec.md the default is `string`

> If no type is indicated, the type is string.

**Comment by alexejk** [comment]
Thats a good catch! then I don't need an extra flag for that. I will adjust the behavior, as this is clearly then a bug on my side.

**Comment by graugans** [comment]
Sorry for the late feedback, with https://github.com/alexejk/go-xmlrpc/releases/tag/v0.4.1 and the `any` type Array it works like a charm. Thanks for your support and great library.
